src/apphosting/githubConnections.ts

```diff
@@ -90,13 +90,26 @@ const ADD_ACCOUNT_CHOICE = "@ADD_ACCOUNT";
 const MANAGE_INSTALLATION_CHOICE = "@MANAGE_INSTALLATION";
 
 /**
- * Prompts the user to link their backend to a GitHub repository.
+ * Prompts the user to create a GitHub connection.
  */
-export async function linkGitHubRepository(
+export async function getOrCreateGithubConnection(
   projectId: string,
   location: string,
-): Promise<devConnect.GitRepositoryLink> {
+  createConnectionId?: string,
+): Promise<devConnect.Connection> {
   utils.logBullet(clc.bold(`${clc.yellow("===")} Import a GitHub repository`));
+
+  if (createConnectionId) {
```

Comment:
would be great to add a unittest for this

Reply:
Done

Reply:
Also added tests for the linkGitHubRepository without a named connection, as they were missing